src/figuresClasses.ts

```diff
@@ -1,19 +1,79 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: string,
+  color: string,
+  getArea(): number,
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 || this.b <= 0 || this.c <= 0
```

Comment:
You don't need to use `this` inside of the constructor. The same all over the code.
```suggestion
    if (a <= 0 || b <= 0 || c <= 0
```